libcommon/strjoin.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * strjoin.c - concatenates strings from an array into a single one
+ *
+ * Mildly inspired by Go standard library's strings.Join function,
+ * but a little bit more rigorous to use since, well, this is C.
+ *
+ * Copyright (c) 2025 Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+char *strjoin(char *elems[], char *sep) {
+ 	size_t en = 0,
+	       joinlen = 0;
+	char *joinbuf = NULL,
+	     *joinbufp = NULL;
+
+	/*
+	 * Count strings in the array using 'NULL' as a sentinel.
+	 * Also already compute the length of each string in
+	 * the array if possible so we can save a loop later.
+	 */
+	for (; elems[en] != NULL; en++)
```

Comment:
Please consider the following:
```
/* Instead of 0, it is initialied with 1 because of the null byte */
size_t joinlen = 1;
char **elem = NULL;

[...]

for (elem = elems; *elem; elem++)
  joinlen += strlen(*elem) + 1;
```
This should be better than everyting until line 37.

Reply:
> I'd like to keep the same logic for calculating the separator. Would it be possible?

Shouldn't be a problem